app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetExercise.java

```diff
@@ -1,21 +1,452 @@
+                                    TimeButton exerciseStartTime, TimeButton exerciseEndTime, String txtExerciseName) {
+        String txtDescription = Objects.requireNonNull(exerciseDescription.getEditText())
+            .getText().toString();
+        String date = exerciseDate.getText().toString();
+        String startHour = exerciseStartTime.getText().toString();
+        String endHour = exerciseEndTime.getText().toString();
+        DateTime startExerciseDateTime = DateTime.Builder.buildDateTimeString(date, startHour);
+        DateTime endExerciseDateTime = DateTime.Builder.buildDateTimeString(date, endHour);
+
+        InfoPetFragment.getCommunication().addExercise(InfoPetFragment.getPet(), txtExerciseName,
+            txtDescription, startExerciseDateTime, endExerciseDateTime);
+        dialog.dismiss();
+    }
+
+    /**
+     * Determines if the exercise parameters are valid.
+     * @param exerciseName The name of the exercise
+     * @param exerciseDate The date of the exercise
+     * @param exerciseStartTime The start time of the exercise
+     * @param exerciseEndTime The end time of the exercise
+     * @param txtExerciseName The name of the exercise
+     * @return True if the parameters are valid
+     */
+    private static boolean isValid(TextInputLayout exerciseName, DateButton exerciseDate, TimeButton exerciseStartTime,
+                                   TimeButton exerciseEndTime, String txtExerciseName) {
+        boolean isValid = true;
+
+        if ("".equals(txtExerciseName)) {
+            exerciseName.setErrorEnabled(true);
+            exerciseName.setError(resources.getString(R.string.error_empty_input_field));
+            isValid = false;
+        }
+
+        if (exerciseDate.getText().toString().equals(resources.getString(R.string.exercise_date))) {
+            exerciseDate.setText(R.string.error_empty_input_field);
+            isValid = false;
+        }
+
+        if (exerciseStartTime.getText().toString().equals(resources.getString(R.string.exercise_start_hour))) {
+            exerciseStartTime.setText(R.string.error_empty_input_field);
+            isValid = false;
+        }
+
+        if (exerciseEndTime.getText().toString().equals(resources.getString(R.string.exercise_end_hour))) {
+            exerciseEndTime.setText(R.string.error_empty_input_field);
+            isValid = false;
+        }
+
+        if (isValid) {
+            isValid = checkDateTimePeriod(exerciseDate, exerciseStartTime, exerciseEndTime);
+        }
+
+        return isValid;
+    }
+
+    /**
+     * Check whether the exercise period is valid.
+     * @param exerciseDate The exercise date
+     * @param exerciseStartTime The exercise start time
+     * @param exerciseEndTime The exercise end time
+     * @return True if the period is valid
+     */
+    private static boolean checkDateTimePeriod(DateButton exerciseDate, TimeButton exerciseStartTime,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method length is 16 lines (max allowed is 15).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5483188)